### ISSUE TITLE
A11y: confusing Back link with new stats box

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#6585]
## Package
- lib-user (I think.)

## Describe the bug
There's a link labelled 'Back' on the new stats page, but it goes to the home page, rather than the previous page.

https://github.com/user-attachments/assets/f0037736-e5ac-42eb-adb0-b3a2f32aa687

## To Reproduce
1. Go to any project Classify page.
2. Click on 'See More' in the stats box.
3. Click on 'Back' above the stats graph.

## Expected behavior
I'd expect a link labelled 'Back' to take me back one page. This link should be labelled 'Home' or something similar.

## Additional context
[WCAG guideline 3.2: Predictable](https://www.w3.org/WAI/WCAG22/Understanding/predictable) is probably relevant here.

**Comment by seanmiller26** [comment]
I agree this functionality is confusing, but this is not a bug but an implementation decision. The button is mistakingly labeled as "Back" where it should have been labeled “Back to my profile” and hard-coded to return users to their logged-in homepage. While there was some debate around whether it should mirror browser back behavior, the team ultimately chose a fixed destination.

**Comment by seanmiller26** [comment]
Agreed!
